### PR TITLE
PGD - Upgrade section reorganised and upgrading with TPA section added

### PR DESCRIPTION
**Pull request by djw-m**
## What Changed?

Reorg, new TPA section, some renaming to help discovery.

### CODE REVIEW

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+In general, you can upgrade the cluster with almost zero downtime by
+using an approach called rolling upgrade. Using this approach, nodes are upgraded one by one, and
+the application connections are switched over to already upgraded nodes.
+
+You can also stop all nodes, perform the upgrade on all nodes, and
+only then restart the entire cluster. This approach is the same as with a standard PostgreSQL setup.
+This strategy of upgrading all nodes at the same time avoids running with
+mixed versions of software and therefore is the simplest. However, it incurs
+some downtime and we don't recommend it unless you can't perform the rolling upgrade
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:16]: Let's be realistic here:
```suggestion
downtime and we don't recommend it unless you can't perform the rolling upgrade
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+LCRs.
+
+A rolling upgrade starts with a cluster with all nodes at a prior release.
+It then proceeds by upgrading one node at a time to the newer release, until
+all nodes are at the newer release. More than two versions
+of any component must never be running at the same time. This requirement means that you must not start the new upgrade until the previous upgrade process fully finishes on all nodes.
+
+An upgrade process can take more time when 
+caution is required to reduce business risk. However, we don't recommend
+running the mixed versions of the software indefinitely.
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:61]:
```suggestion
running mixed versions of the software indefinitely.
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+While the cluster is going through a rolling upgrade, mixed versions of software
+are running in the cluster. For example, suppose nodeA has PGD 3.7.16, while
+nodeB and nodeC has 4.1.0. In this state, the replication and group
+management uses the protocol and features from the oldest version (3.7.16
+in this example), so any new features provided by the newer version
+that require changes in the protocol are disabled. Once all nodes are
+upgraded to the same version, the new features are enabled.
+
+Similarly, when a cluster with WAL-decoder-enabled nodes is going through a
+rolling upgrade, WAL decoder on a higher version of PGD node produces LCRs
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:46]: As this is the first time we mention it on this page, how about explaining what "LCR" stands for?

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+produces LCRs with lower pglogical version. As a result, WAL senders on a higher
+version of PGD nodes are not expected to use LCRs due to a mismatch in protocol
+versions. On a lower version of PGD nodes, WAL senders may continue to use
+LCRs. Once all the PGD nodes are on the same PGD version, WAL senders use
+LCRs.
+
+A rolling upgrade starts with a cluster with all nodes at a prior release.
+It then proceeds by upgrading one node at a time to the newer release, until
+all nodes are at the newer release. More than two versions
+of any component must never be running at the same time. This requirement means that you must not start the new upgrade until the previous upgrade process fully finishes on all nodes.
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:57]:
```suggests
all nodes are at the newer release. At most, two versions
of any component can be running simultaneously. This restriction means you can only start a new upgrade once the previous upgrade process fully finishes on all nodes.
```

Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:57]:
```suggestion
all nodes are at the newer release. At most, two versions
of any component can be running simultaneously. This restriction means you can only start a new upgrade once the previous upgrade process fully finishes on all nodes.
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+### Rolling server software upgrades
+
+A rolling upgrade is the process where the [server
+software upgrade](#server-software-upgrade) process is performed on each node in the
+cluster one after another while keeping the rest of the cluster
+operating.
+
+The actual procedure depends on whether the Postgres component is being
+upgraded to a new major version.
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:80]: Lots of passive voice here: "is performed", "is being upgraded" - could you improve this?

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+The other method to upgrade the server software is to join a new node
+to the cluster and later drop one of the existing nodes running
+the older version of the software.
+
+For this approach, the procedure is always the same. However, because it
+includes node join, a potentially large data transfer is required.
+
+Take care not to use features that are available only in
+the newer Postgres versions until all nodes are upgraded to the
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:96]:
```suggestion
the newer Postgres version until all nodes are upgraded to the
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    You can't use it for rolling upgrades by way of joining a new node method. Instead, use a logical join.
+
+### Upgrading a CAMO-enabled cluster
+
+Upgrading a CAMO-enabled cluster requires upgrading CAMO groups one by one while
+disabling the CAMO protection for the group being upgraded and reconfiguring it
+using the new [commit scope](../durability/commit-scopes)-based settings.
+
+The following approach is recommended for upgrading two BDR nodes that
+constitute a CAMO pair to PGD 5.0:
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:114]:
```suggestion
We recommend the following approach for upgrading two BDR nodes that
constitute a CAMO pair to PGD 5.0:
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+However, sometimes more steps, like reindexing, might be recommended for
+specific minor version upgrades. Refer to the release notes of the 
+version of Postgres you're upgrading to.
+
+#### Major version Postgres upgrade
+
+Upgrading to a new major version of Postgres is more complicated than upgrading to a minor version.
+
+EDB Postgres Distributed provides a bdr_pg_upgrade command line utility,
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:189]:
```suggestion
EDB Postgres Distributed provides a `bdr_pg_upgrade` command line utility,
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+## Upgrade check and validation
+
+After you upgrade your PGD node, you can verify the current
+version of the binary:
+
+```sql
+SELECT bdr.bdr_version();
+```
+
+Always check the [monitoring](../monitoring) after upgrading a node to confirm
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:206]:
```suggestion
Always check your [monitoring](../monitoring) application after upgrading a node to confirm
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+that the upgraded node is working as expected.
+
+## Moving from HARP to PGD Proxy
+
+HARP can for a time coexist with the new
+[connection management](../routing) configuration. This means you can:
```
Comment (LenzGr) [inline product_docs/docs/pgd/5/upgrades/manual_overview.mdx:212]:
```suggestion
HARP can coexist with the new
[connection management](../routing) configuration temporarily. This means you can:
```

File: product_docs/docs/pgd/5/upgrades/manual_overview.mdx (hunk 3 above)
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/manual_overview.mdx:46]:
```suggestion
rolling upgrade, WAL decoder on a higher version of PGD node produces LCRs ([logical change records](..//node_management/decoding_worker/#enabling)
```

File: product_docs/docs/pgd/5/upgrades/tpa_overview.mdx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+## Upgrading
+
+Run
+
+```
+tpaexec upgrade clustername
+```
+
+Where clustername is the name of the cluster and the path to the cluster configuration directory. By default, TPA will upgrade each node of the cluster to the latest versions of the software the nodes were configured with.
```
[the remark this answers is not in the document]
Comment (djw-m) [reply product_docs/docs/pgd/5/upgrades/tpa_overview.mdx:23]: No using code marking where it isn't code. I'll mark up clustername, but as bold.

**Review by sonotley — APPROVED** [review]

**Pull request merged**